src/components/Overlay.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import styled from "styled-components";
+
+type overlayProps = {
+  onCloseModal: React.Dispatch<React.SetStateAction<boolean>>; //sets state modalOpen = false;
+  children: React.ReactNode;
+};
+
+export const BackDropStyle = styled.div`
```

Comment:
There should be a transition

Reply:
Open transition is done. Closing transition I need some time to research. Seems more complicated.

Reply:
Check out this library https://reactcommunity.org/react-transition-group/transition

Reply:
you can't really animate out because when the menu is closed the entire element is removed from the DOM immediately, so either you code the DOM removal after the animation or you use a library to do it for you